server/src/main/proto/spine/server/entity/entity.proto

```diff
@@ -48,19 +48,28 @@ message EntityRecord {
     // The version of the entity.
     base.Version version = 3;
 
-    // The visibility flags of the entity.
-    Visibility visibility = 4;
+    // The lifecycle flags of the entity.
+    LifecycleFlags lifecycle_flags = 4;
 }
 
-// The visibility flags of an entity.
-//
-// If one of the flags is set to `true` the entity becomes “invisible” to regular queries.
-//
-message Visibility {
+// The lifecycle flags of an entity.
+message LifecycleFlags {
```

Comment:
It would require having a general mechanism of transforming a message into a set of storage fields. And detecting simple cases like just `boolean` flags like this. Kinda Hibernate :) I'd say, it should be a separate PR anyway.